cmd/agent/app/reporter/http/reporter.go

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2017 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+
+	"github.com/uber/jaeger/thrift-gen/jaeger"
+	"github.com/uber/jaeger/thrift-gen/zipkincore"
+)
+
+const (
+	jaegerBatches = "jaeger"
+	zipkinBatches = "zipkin"
+)
+
+type batchMetrics struct {
+	// Number of successful batch submissions to collector
+	BatchesSubmitted metrics.Counter `metric:"batches.submitted"`
+
+	// Number of failed batch submissions to collector
+	BatchesFailures metrics.Counter `metric:"batches.failures"`
+
+	// Number of spans in a batch submitted to collector
+	BatchSize metrics.Gauge `metric:"batch_size"`
+
+	// Number of successful span submissions to collector
+	SpansSubmitted metrics.Counter `metric:"spans.submitted"`
+
+	// Number of failed span submissions to collector
+	SpansFailures metrics.Counter `metric:"spans.failures"`
+}
+
+var httpClient = &http.Client{Timeout: 2 * time.Second}
+
+// Reporter forwards received spans to central collector tier over plain HTTP.
+type Reporter struct {
+	scheme             string   // http or https
+	collectorHostPorts []string // only the first value is used
+
+	authToken string
+	username  string
+	password  string
```

Comment:
is this really a possibility that someone will be running the agent with basic auth?

Reply:
I might be missing something. I understand that this reporter is to be used for the communication between the Agent and Collector. This is not for protecting the agent.